src/compiler/sys.ts

```diff
@@ -1413,8 +1424,7 @@ namespace ts {
             }
 
             function getDirectories(path: string): string[] {
-                perfLogger.logEvent("ReadDir: " + path);
-                return filter<string>(_fs.readdirSync(path), dir => fileSystemEntryExists(combinePaths(path, dir), FileSystemEntryKind.Directory));
+                return getAccessibleFileSystemEntries(path).directories.slice();

```

Comment:
You don't need to slice since getAccessibleFileSystemEntries returns new array for directories..

Reply:
Not always: `return emptyFileSystemEntries` does not. Of course, that’s easy to change. I’ve added a commit doing so.